src/mainwindow.cpp

```diff
@@ -38,7 +38,7 @@ MainWindow::MainWindow(QWidget *parent) :
     connect(app->getAction(KiwixApp::KiwixServeAction), &QAction::triggered,
             mp_localKiwixServer, &QDialog::show);
 
-    connect(app, &KiwixApp::currentTitleChanged, this, [=](const QString& title) {
+    connect(mp_ui->tabBar, &TabBar::currentTitleChanged, this, [=](const QString& title) {
         if (!title.isEmpty() && !title.startsWith("zim://"))
             setWindowTitle(title + " - Kiwix");
         else
```

Comment:
There are three separate `connect(mp_ui->tabBar, &TabBar::currentTitleChanged, ...)` statements in the `MainWindow` constructor. Isn't it better to introduce a new private slot `MainWindow::onCurrentTabTitleChanged()` and localize the handling of that event entirely in one function?

Reply:
I incline to leave it as separate 3 signal-slot connections.